### PR TITLE
feat: add new trip

### LINKED ISSUE #72: feat: setup `Add New Trip` button
- Add the button in `/`
- Should be rendered only when the user is signed in
- Use dialog button -> https://tailwindcss-radix.vercel.app/#dialog

1. Trip Name
2. Trip Description
3. Start Date
4. End Date

### PR DESCRIPTION
**Pull request by ssarkar551**

### CODE REVIEW

**Comment by ssarkar551** [comment]
added new trip button 
image

File: src/components/NewTripButton.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
+import { Transition } from "@headlessui/react";
```
Comment (AyanavaKarmakar) [inline src/components/NewTripButton.tsx]: - create a new component called Home and move it there `src/components/Home/NewTripButton.tsx`
- add export in `src/components/index.ts`

File: src/components/NewTripButton.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+import { Fragment, useState } from "react";
+import { regularFont } from "../fonts";
+
+export const NewTripButton = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [tripName, setTripName] = useState("");
+  const [tripDesc, setTripDesc] = useState("");
+  const [startDate, setStartDate] = useState("");
+  const [endDate, setEndDate] = useState("");
+  return (
```
Comment (AyanavaKarmakar) [inline src/components/NewTripButton.tsx]: add an extra space between lines 13 & 14. Looks congested rn

File: src/pages/index.tsx [hunk 3]
```diff
@@ -6,6 +6,7 @@ import Link from "next/link";
 import clsx from "clsx";
 import { boldFont } from "../fonts";
 import { api } from "../utils/api";
+import { NewTripButton } from "../components/NewTripButton";
```
Comment (AyanavaKarmakar) [inline src/pages/index.tsx]: update import in line 2

```suggestion
```

File: src/components/NewTripButton.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+                    )}
+                  />
+                </fieldset>
+                <fieldset>
+                  <label htmlFor="startDate" className="text-lg">
+                    Start Date
+                  </label>
+                  <input
+                    id="startDate"
+                    type="date"
```
Comment (AyanavaKarmakar) [inline src/components/NewTripButton.tsx]:
```suggestion
                    type="datetime-local"
```

File: src/components/NewTripButton.tsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+                    )}
+                  />
+                </fieldset>
+                <fieldset>
+                  <label htmlFor="endDate" className="text-lg">
+                    End Date
+                  </label>
+                  <input
+                    id="endDate"
+                    type="date"
```
Comment (AyanavaKarmakar) [inline src/components/NewTripButton.tsx]:
```suggestion
                    type="datetime-local"
```

File: src/components/NewTripButton.tsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+              forceMount
+              className={clsx(
+                `${regularFont.className}`,
+                "fixed z-50",
+                "w-[95vw] max-w-md rounded-lg p-6 md:w-full",
+                "top-[50%] left-[50%] -translate-x-[50%] -translate-y-[50%]",
+                "bg-gray-100 dark:bg-gray-800",
+              )}
+            >
+              <Dialog.Title className="text-2xl">New Trip</Dialog.Title>
```
Comment (AyanavaKarmakar) [inline src/components/NewTripButton.tsx]:
```suggestion
              <Dialog.Title className="text-2xl">Create New Trip</Dialog.Title>
```

File: src/components/NewTripButton.tsx [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+              <Dialog.Description className="mt-2 text-xl">
+                Click here to set up a new trip
```
Comment (AyanavaKarmakar) [inline src/components/NewTripButton.tsx]:
```suggestion
                Fill this form to set up a new trip
```

File: src/components/NewTripButton.tsx [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+                    )}
+                  />
+                </fieldset>
+                <fieldset>
+                  <label htmlFor="email" className="text-lg">
+                    Trip Description
+                  </label>
+                  <input
+                    id="tripDesc"
+                    type="textarea"
```
Comment (AyanavaKarmakar) [inline src/components/NewTripButton.tsx]:
```suggestion
```

File: src/components/NewTripButton.tsx [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+                      "text-xl",
+                      "border border-gray-400 focus-visible:border-transparent dark:border-gray-700 dark:bg-gray-800",
+                    )}
+                  />
+                </fieldset>
+                <fieldset>
+                  <label htmlFor="email" className="text-lg">
+                    Trip Description
+                  </label>
+                  <input
```
Comment (AyanavaKarmakar) [inline src/components/NewTripButton.tsx]:
```suggestion
                  <textarea
```

File: src/components/NewTripButton.tsx [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+                    id="tripDesc"
```
Comment (AyanavaKarmakar) [inline src/components/NewTripButton.tsx]:
```suggestion
                    id="tripDescription"
```

File: src/pages/index.tsx [hunk 11]
```diff
@@ -43,6 +44,7 @@ const Home: NextPage = () => {
         )}
       >
         {/** TODO - Placehold Banner. Should be removed later. */}
+        {status === "authenticated" && <NewTripButton />}
```
Comment (AyanavaKarmakar) [inline src/pages/index.tsx]: move this to the line below </Link>

File: src/components/NewTripButton.tsx [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+import * as Dialog from "@radix-ui/react-dialog";
+import { CrossIcon } from "../icons";
+import clsx from "clsx";
+import { Fragment, useState } from "react";
+import { regularFont } from "../fonts";
+
+export const NewTripButton = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [tripName, setTripName] = useState("");
+  const [tripDesc, setTripDesc] = useState("");
```
Comment (AyanavaKarmakar) [inline src/components/NewTripButton.tsx]:
```suggestion
  const [tripDescription, setTripDescription] = useState("");
```

**Review by AyanavaKarmakar — CHANGES_REQUESTED** [review]

**Comment by AyanavaKarmakar** [comment]
image

image

**Review by AyanavaKarmakar — APPROVED** [review]

**Pull request merged**